htdocs/projet/class/project.class.php

```diff
@@ -2549,4 +2549,66 @@ public function getChildren()
 
 		return $children;
 	}
+
+	/**
+	 * method for calcule weekly time spent and get a repport
+	 * @return string|int
+	 */
+	public function createWeeklyReport()
+	{
+		global $mysoc;
+		$lastWeekStart = new DateTime();
+		$lastWeekStart->modify('last monday -7 days');
```

Comment:
first day of week may vary depending on user setup.
Can you check if you can replace with dol_get_first_day_week() in date.lib.php